src/wp-includes/class-wp-plugin-dependencies.php

```diff
@@ -696,8 +696,16 @@ protected static function get_dependency_filepaths() {
 	 * Retrieves and stores dependency plugin data from the WordPress.org Plugin API.
 	 *
 	 * @since 6.5.0
+	 *
+	 * @global string $pagenow The filename of the current screen.
 	 */
 	protected static function get_dependency_api_data() {
+		global $pagenow;
+
+		if ( ! is_admin() || ( 'plugins.php' !== $pagenow && 'plugin-install.php' !== $pagenow ) ) {
```

Comment:
Would it be better to add the pages to an array and the check `! in_array()`?

Reply:
string checks are faster than `in_array()`, and it also saves on  creating an array in memory.

I've also been thinking of removing the default value I'm supplying in `get_option( 'plugin_data', array() );` in recent PRs for the same reason. I'll follow up with a PR for this later/tomorrow